src/lib.rs

```diff
@@ -382,30 +382,208 @@ impl<Types: IpfsTypes> Ipfs<Types> {
         self.repo.get_block(cid).instrument(self.span.clone()).await
     }
 
-    /// Remove block from the ipfs repo.
+    /// Remove block from the ipfs repo. A pinned block cannot be removed.
     pub async fn remove_block(&self, cid: Cid) -> Result<Cid, Error> {
         self.repo
             .remove_block(&cid)
             .instrument(self.span.clone())
             .await
     }
 
-    /// Pins a given Cid
-    pub async fn pin_block(&self, cid: &Cid) -> Result<(), Error> {
-        self.repo.pin_block(cid).instrument(self.span.clone()).await
+    /// Pins a given Cid recursively or directly (non-recursively).
+    ///
+    /// Pins on a block are additive in sense that a previously directly (non-recursively) pinned
+    /// can be made recursive, but removing the recursive pin on the block removes also the direct
+    /// pin as well.
+    ///
+    /// Pinning a Cid recursively (for supported dag-protobuf and dag-cbor) will walk it's
+    /// references and pin the references indirectly. When a Cid is pinned indirectly it will keep
+    /// it's previous direct or recursive pin and be indirect in addition.
+    ///
+    /// Recursively pinned Cids cannot be pinned non-recursively but non-recursively pinned Cids
```

Comment:
```suggestion
    /// Recursively pinned Cids cannot be re-pinned non-recursively but non-recursively pinned Cids
```
as long as am I understanding correctly that they can't be pinned again that way

Reply:
Not sure if this would be better as in my mind, the pin stays all of the time, but is modified.

Reply:
it's a more linguistical concern than anything; consider the following re-ordering of that phrase: "Cids pinned recursively cannot be pinned non-recursively" - it's just a bit of a grammatical conundrum